isabelle/Core/Semantics.thy

```diff
@@ -721,34 +721,80 @@ and maxTimeCase :: "Case \<Rightarrow> int" where
 
 subsection "calculateNonAmbiguousInterval"
 
+text "Helper functions for \<^emph>\<open>calculateNonAmbiguousInterval\<close>"
 fun gtIfNone :: "int option \<Rightarrow> int \<Rightarrow> bool" where
 "gtIfNone None _ = True" |
 "gtIfNone (Some x) y = (x > y)"
 
 fun geIfNone :: "int option \<Rightarrow> int \<Rightarrow> bool" where
 "geIfNone None _ = True" |
-"geIfNone (Some x) y = (x \<ge> y)"
+"geIfNone (Some x) y = (x \<ge> y) "
 
 fun subIfSome :: "int option \<Rightarrow> int \<Rightarrow> int option" where
 "subIfSome None _ = None" |
 "subIfSome (Some x) y = Some (x - y)"
 
 
+text \<open>
+A TimeInterval (startTime, endTime) can be ambiguous wrt a When's timeout if startTime < timeout \<le> endTime
+
+\<close>
+
+text \<open>The \<^emph>\<open>calculateNonAmbiguousInterval\<close> function can help a user to calculate a TimeInterval that
+won't be ambiguous for the next \<^emph>\<open>n\<close> inputs of a contract.\<close>
+
+text \<open>The only Contract constructor that can yield a \<^term>\<open>TEAmbiguousTimeIntervalError\<close> is the \<^term>\<open>When\<close> case. 
+Computing a transaction of a contract that starts in a non quiescent state (\<^term>\<open>Let\<close>, \<^term>\<open>Pay\<close>, \<^term>\<open>If\<close>, \<^term>\<open>Assert\<close>) 
+can end in a \<^term>\<open>TEAmbiguousTimeIntervalError\<close> iff there is a \<^term>\<open>When\<close> case that makes it ambiguous.
+\<close>
+
+text \<open>A TimeInterval expressed as the tuple \<^term>\<open>(startTime \<times> endTime)\<close> can be ambiguous wrt a \<^term>\<open>When\<close>'s timeout
+iff \<^emph>\<open>startTime < timeout \<le> endTime\<close>
+\<close>
+
+text
+\<open> It parameters of \<^emph>\<open>calculateNonAmbiguousInterval\<close> are:
```

Comment:
```suggestion
\<open> The parameters of \<^emph>\<open>calculateNonAmbiguousInterval\<close> are:
```

Reply:
Thanks, fixing this in the Semantics refactoring as this function is no longer in this file